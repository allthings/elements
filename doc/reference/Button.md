<!-- 
This is an auto-generated markdown. 
You can change it in "src/Button/Button.tsx" and run build:docs to update this file.
-->
# Button
Buttons make common actions immediately visible and easy to perform with one
click or tap. They can be used for any type of action, including navigation.

You can use two different looks for the button: primary and
secondary. Primary is the default type, so there's no need to explicitly
define it.

```example
<ThemeProvider>
   <Button>Hello you</Button>
</ThemeProvider>
```

To have an icon as button-label, just add the icon-component as children.

```example
<Button type="submit">
    <View direction="row">
      Hello with icon
      <View style={{ marginLeft: 10 }}>
        <Icon name="send-filled" size="xs" color="white" />
      </View>
    </View>
  </Button>
```
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|secondary|boolean|If the button is used for a secondary option
|onClick|((event: MouseEvent<Element, MouseEvent>) => void)|
|type|"reset", "button", "submit"|Type of the button (deprecated)
|color|string|
|disabled|boolean|
|backgroundColor|string|Color of the button, theme primary color by default
|name|string|Name of the
