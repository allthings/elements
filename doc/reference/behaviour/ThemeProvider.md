<!-- 
This is an auto-generated markdown. 
You can change it in "src/behaviour/ThemeProvider.jsx" and run build:docs to update this file.
-->
# ThemeProvider
All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
The ThemeProvider allows you to define the default colors for most elements.

**Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|root|bool|
|theme|shape|<br>Default: {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.secondary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}
|children|node|
