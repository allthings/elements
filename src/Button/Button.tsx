import { css } from 'glamor'
import { color as col, lightness } from 'kewler'
import React, { HTMLProps, MouseEvent } from 'react'
import { createTextStyles } from '../Text'
import Theme from '../Theme'
import { color } from '../utils/propTypes/color'

const baseStyle = {
  position: 'relative',
  padding: '10px 16px',
  borderRadius: '2px',
  border: 'none',
  userSelect: 'none',
  outline: 'none',
}

const primary = ({
  color,
  disabled,
  disabledBackgroundColor,
  backgroundColor,
}: {
  color: color
  disabled: boolean
  disabledBackgroundColor: color
  backgroundColor: color
}) => ({
  color,
  background: disabled ? disabledBackgroundColor : backgroundColor,
})

const secondary = ({
  disabled,
  disabledColor,
  backgroundColor,
}: {
  disabled: boolean
  disabledColor: color
  backgroundColor: color
}) => ({
  background: 'transparent',
  color: disabled ? disabledColor : backgroundColor,
  margin: 2,
})

function styles(
  backgroundColor: string,
  color: string,
  isDisabled: boolean,
  isSecondary: boolean,
) {
  const props = {
    color,
    disabled: isDisabled,
    disabledBackgroundColor: 'lightGrey',
    disabledColor: 'darkgray',
    backgroundColor,
  }
  return css({
    ...baseStyle,
    ...(isSecondary ? secondary(props) : primary(props)),
    cursor: isDisabled ? 'not-allowed' : 'pointer',
    transition: '250ms ease-in-out',
    ':focus': {
      outline: 'none',
    },
    ':hover': isSecondary
      ? !isDisabled && { color: col(backgroundColor, lightness(-20)) }
      : {
          color,
          background: isDisabled
            ? props.disabledBackgroundColor
            : backgroundColor.indexOf('#') !== -1
            ? col(backgroundColor, lightness(-10))
            : backgroundColor,
        },
  })
}

interface IButtonProps extends HTMLProps<HTMLButtonElement> {
  /** If the button is used for a secondary option */
  readonly secondary?: boolean
  readonly onClick?: (event: MouseEvent) => void
  /** Type of the button (deprecated) */
  readonly type?: 'reset' | 'button' | 'submit'
  readonly color?: color
  readonly disabled?: boolean
  /** Color of the button, theme primary color by default */
  readonly backgroundColor?: string
  /** Name of the */
  readonly name?: string
}

/**
 * Buttons make common actions immediately visible and easy to perform with one
 * click or tap. They can be used for any type of action, including navigation.
 *
 * You can use two different looks for the button: primary and
 * secondary. Primary is the default type, so there's no need to explicitly
 * define it.
 *
 * ```example
 * <ThemeProvider>
 *   <Button>Hello you</Button>
 * </ThemeProvider>
 * ```
 *
 * To have an icon as button-label, just add the icon-component as children.
 *
 * ```example
 * <Button type="submit">
 *    <View direction="row">
 *      Hello with icon
 *      <View style={{ marginLeft: 10 }}>
 *        <Icon name="send-filled" size="xs" color="white" />
 *      </View>
 *    </View>
 *  </Button>
 * ```
 */
const Button = ({
  children,
  type = 'button',
  disabled = false,
  backgroundColor = 'primary',
  color = 'white',
  secondary = false,
  onClick,
  name,
  ...restProps
}: IButtonProps) => (
  <Theme>
    {({ colorize }) => (
      <button
        type={type}
        {...css(
          styles(
            colorize(backgroundColor),
            colorize(color),
            disabled,
            secondary,
          ),
        )}
        {...restProps}
        {...createTextStyles({ size: 'l' })}
        name={name || type}
        onClick={disabled ? undefined : onClick}
      >
        {children}
      </button>
    )}
  </Theme>
)

export default Button
