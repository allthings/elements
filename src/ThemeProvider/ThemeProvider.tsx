import React, { FunctionComponent, createContext } from 'react'
import { ColorPalette } from '@allthings/colors'

export const defaultTheme: ITheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.secondary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

export interface ITheme {
  readonly primary: string
  readonly text: string
  readonly secondaryText: string
  readonly titleColor: string
  readonly contrast: string
  readonly warn: string
  readonly disabled: string
  readonly background: string
  readonly textOnBackground: string
  readonly [key: string]: string
}

/**
 * All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
 * The ThemeProvider allows you to define the default colors for most elements.
 *
 * **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
 **/
const ThemeProvider: FunctionComponent<{ theme?: Partial<ITheme> }> = ({
  children,
  theme = defaultTheme,
}) => (
  <ThemeConsumer>
    {contextTheme => (
      <ThemeContext.Provider value={{ ...contextTheme, ...(theme as ITheme) }}>
        {children}
      </ThemeContext.Provider>
    )}
  </ThemeConsumer>
)

export default ThemeProvider
export const ThemeContext = createContext<ITheme>(defaultTheme)
export const ThemeConsumer = ThemeContext.Consumer
