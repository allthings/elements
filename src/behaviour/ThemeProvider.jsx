import React from 'react'
import PropTypes from 'prop-types'
import { Broadcast, Subscriber } from 'react-broadcast'
import { ColorPalette } from '@allthings/colors'

export const THEME_CHANNEL = 'theme'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.primary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

/**
* All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
* The ThemeProvider allows you to define the default colors for most elements.
*
* **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
**/
export default class ThemeProvider extends React.PureComponent {
  static propTypes = {
    root: PropTypes.bool,
    theme: PropTypes.shape({
      primary: PropTypes.string,
      text: PropTypes.string,
      secondaryText: PropTypes.string,
      titleColor: PropTypes.string,
      contrast: PropTypes.string,
      warn: PropTypes.string,
      disabled: PropTypes.string,
      background: PropTypes.string,
      textOnBackground: PropTypes.string,
    }),
    children: PropTypes.node,
  }

  static defaultProps = {
    theme: defaultTheme,
  }

  static contextTypes = Subscriber.contextTypes

  render() {
    let root = true
    if (this.context.broadcasts && this.context.broadcasts[THEME_CHANNEL]) {
      root = false
    }
    const { theme, children } = this.props
    return root ? (
      <Broadcast
        channel={THEME_CHANNEL}
        value={{
          ...ThemeProvider.defaultProps.theme,
          ...theme,
        }}
      >
        {children}
      </Broadcast>
    ) : (
      <Subscriber channel={THEME_CHANNEL}>
        {previousTheme => (
          <Broadcast
            channel={THEME_CHANNEL}
            value={{
              ...ThemeProvider.defaultProps.theme,
              ...previousTheme,
              ...theme,
            }}
          >
            {children}
          </Broadcast>
        )}
      </Subscriber>
    )
  }
}

export const ThemeShape = ThemeProvider.propTypes.theme

export const withTheme = (mapThemeToProps, displayName) => WrappedComponent =>
  // eslint-disable-next-line
  class extends React.PureComponent {
    static displayName = displayName || WrappedComponent.displayName
    static component = WrappedComponent

    renderComponent = theme => {
      const props = !mapThemeToProps
        ? { theme }
        : mapThemeToProps(theme, this.props)

      return <WrappedComponent {...this.props} {...props} />
    }

    render() {
      return (
        <Subscriber channel={THEME_CHANNEL}>{this.renderComponent}</Subscriber>
      )
    }
  }
