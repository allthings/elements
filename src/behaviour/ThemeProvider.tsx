import React, { FunctionComponent, createContext } from 'react'
import { ColorPalette } from '@allthings/colors'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.secondary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

export interface ITheme {
  primary: string
  text: string
  secondaryText: string
  titleColor: string
  contrast: string
  warn: string
  disabled: string
  background: string
  textOnBackground: string
  [key: string]: string
}

const ThemeContext = createContext<ITheme>(defaultTheme)
export const ThemeConsumer = ThemeContext.Consumer

/**
 * All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
 * The ThemeProvider allows you to define the default colors for most elements.
 *
 * **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
 **/
const ThemeProvider: FunctionComponent<{ theme?: Partial<ITheme> }> = ({
  children,
  theme = {},
}) => (
  <ThemeConsumer>
    {contextTheme => (
      <ThemeContext.Provider value={{ ...contextTheme, ...(theme as ITheme) }}>
        {children}
      </ThemeContext.Provider>
    )}
  </ThemeConsumer>
)

export default ThemeProvider

type mapThemeToPropsType = (theme: ITheme, props: any) => any

function createThemeConsumer<T>(
  mapThemeToProps: mapThemeToPropsType,
  WrappedComponent: React.ComponentType<T>,
  props: Omit<T, 'theme'>
) {
  return (theme: ITheme) => {
    const extraProps = !mapThemeToProps
      ? { theme }
      : mapThemeToProps(theme, props)

    return <WrappedComponent {...props} {...extraProps} />
  }
}

export function withTheme(mapThemeToProps: mapThemeToPropsType) {
  return function<T extends { theme: ITheme }>(
    WrappedComponent: React.ComponentType<T>
  ): React.ComponentType<Omit<T, 'theme'>> {
    return function(props: Omit<T, 'theme'>) {
      return (
        <ThemeConsumer>
          {createThemeConsumer<T>(mapThemeToProps, WrappedComponent, props)}
        </ThemeConsumer>
      )
    }
  }
}
