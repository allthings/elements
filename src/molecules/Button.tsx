import React, { HTMLProps, MouseEvent, FunctionComponent } from 'react'
import { css } from 'glamor'
import { createTextStyles } from '../atoms/Text'
import { color } from '../propTypes/color'
import { color as col, lightness } from 'kewler'
import Theme from '../behaviour/Theme'

const baseStyle = {
  position: 'relative',
  padding: '10px 16px',
  borderRadius: '2px',
  border: 'none',
  userSelect: 'none',
  outline: 'none',
}

const primary = ({
  color,
  disabled,
  disabledBackgroundColor,
  backgroundColor,
}: {
  color: color
  disabled: boolean
  disabledBackgroundColor: color
  backgroundColor: color
}) => ({
  color,
  background: disabled ? disabledBackgroundColor : backgroundColor,
})

const secondary = ({
  disabled,
  disabledColor,
  backgroundColor,
}: {
  disabled: boolean
  disabledColor: color
  backgroundColor: color
}) => ({
  background: 'transparent',
  color: disabled ? disabledColor : backgroundColor,
  margin: 2,
})

function styles(
  backgroundColor: string,
  color: string,
  isDisabled: boolean,
  isSecondary: boolean,
) {
  const props = {
    color,
    disabled: isDisabled,
    disabledBackgroundColor: 'lightGrey',
    disabledColor: 'darkgray',
    backgroundColor,
  }
  return css({
    ...baseStyle,
    ...(isSecondary ? secondary(props) : primary(props)),
    cursor: isDisabled ? 'not-allowed' : 'pointer',
    transition: '250ms ease-in-out',
    ':focus': {
      outline: 'none',
    },
    ':hover': isSecondary
      ? !isDisabled && { color: col(backgroundColor, lightness(-20)) }
      : {
          color,
          background: isDisabled
            ? props.disabledBackgroundColor
            : backgroundColor.indexOf('#') !== -1
            ? col(backgroundColor, lightness(-10))
            : backgroundColor,
        },
  })
}

interface IButtonProps extends HTMLProps<HTMLButtonElement> {
  /** If the button is used for a secondary option */
  secondary?: boolean
  onClick?: (event: MouseEvent) => void
  /** Type of the button (deprecated) */
  type?: 'reset' | 'button' | 'submit'
  color?: color
  disabled?: boolean
  /** Color of the button, theme primary color by default */
  backgroundColor?: string
  /** Name of the */
  name?: string
}

/**
 * Buttons make common actions immediately visible and easy to perform with one
 * click or tap. They can be used for any type of action, including navigation.
 *
 * You can use two different looks for the button: primary and
 * secondary. Primary is the default type, so there's no need to explicitly
 * define it.
 *
 * ```example
 * <ThemeProvider>
 *   <Button>Hello you</Button>
 * </ThemeProvider>
 * ```
 *
 * To have an icon as button-label, just add the icon-component as children.
 *
 * ```example
 * <Button type="submit">
 *    <View direction="row">
 *      Hello with icon
 *      <View style={{ marginLeft: 10 }}>
 *        <Icon name="send-filled" size="xs" color="white" />
 *      </View>
 *    </View>
 *  </Button>
 * ```
 */
const Button: FunctionComponent<IButtonProps> = ({
  children,
  type = 'button',
  disabled = false,
  backgroundColor = 'primary',
  color = 'white',
  secondary = false,
  onClick,
  name,
  ...restProps
}) => (
  <Theme>
    {({ colorize }) => (
      <button
        type={type}
        {...css(
          styles(
            colorize(backgroundColor),
            colorize(color),
            disabled,
            secondary,
          ),
        )}
        {...restProps}
        {...createTextStyles({ size: 'l' })}
        name={name || type}
        onClick={disabled ? undefined : onClick}
      >
        {children}
      </button>
    )}
  </Theme>
)

export default Button
