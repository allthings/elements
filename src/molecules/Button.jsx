import PropTypes from 'prop-types'
import React from 'react'
import { css } from 'glamor'
import Text from '../atoms/Text'
import { withTheme } from '../behaviour/ThemeProvider'
import { color, colorCode } from '../propTypes/color'
import { color as col, lightness } from 'kewler'

const baseStyle = {
  position: 'relative',
  padding: '8px 14px',
  borderRadius: '2px',
  userSelect: 'none',
  outline: 'none',
  border: 'none',
}

function styles(
  backgroundColor,
  color,
  disabled,
  disabledColor,
  disabledBackgroundColor
) {
  return css({
    ...baseStyle,
    background: disabled ? disabledBackgroundColor : backgroundColor,
    color: disabled ? disabledColor : color,
    cursor: disabled ? 'not-allowed' : 'pointer',
    transition: '250ms ease-in-out',
    ':focus': {
      outline: 'none',
    },
    ':hover': {
      background: disabled
        ? disabledBackgroundColor
        : backgroundColor.includes('rgb')
        ? backgroundColor
        : col(backgroundColor, lightness(-10)),
    },
  })
}

/**
 * Buttons make common actions immediately visible and easy to perform with one
 * click or tap. They can be used for any type of action, including navigation.
 *
 * You can use two different looks for the button: primary and
 * secondary. Primary is the default type, so there's no need to explicitly
 * define it.
 *
 * ```example
 * <ThemeProvider>
 *   <Button>Hello you</Button>
 * </ThemeProvider>
 * ```
 *
 * To have an icon as button-label, just add the icon-component as children.
 *
 * ```example
 * <Button type="submit">
 *    <View direction="row">
 *      Hello with icon
 *      <View style={{ marginLeft: 10 }}>
 *        <Icon name="send-filled" size="xs" color="white" />
 *      </View>
 *    </View>
 *  </Button>
 * ```
 */
class Button extends React.Component {
  static propTypes = {
    /** Just text most of the time */
    children: PropTypes.node.isRequired,
    /** Called when the button is clicked */
    onClick: PropTypes.func,
    /** Type of the button (deprecated) */
    type: PropTypes.oneOf(['reset', 'button', 'submit']),
    /** Disable button state to indicate it's not touchable */
    disabled: PropTypes.bool,
    /** Color of the button, theme primary color by default */
    backgroundColor: PropTypes.string,
    /** Textcolor of the button (deprecated) */
    color: color,
    /** Textcolor when button is disabled (deprecated) */
    disabledColor: PropTypes.string,
    /** Color when button is disabled (deprecated) */
    disabledBackgroundColor: PropTypes.string,
    /** Pass your own css (deprecated) */
    css: PropTypes.object,
  }

  handleClick = e => {
    if (!this.props.disabled) {
      this.props.onClick && this.props.onClick(e)
    } else {
      e.preventDefault()
    }
  }

  render() {
    const {
      children,
      type,
      disabled,
      backgroundColor,
      color,
      disabledColor,
      disabledBackgroundColor,
      css: cssProp,
      ...restProps
    } = this.props

    const allStyles = css(
      styles(
        colorCode(backgroundColor),
        color,
        disabled,
        disabledColor,
        disabledBackgroundColor
      ),
      cssProp
    )

    return (
      <button
        type={type}
        {...allStyles}
        {...restProps}
        name={restProps.name || type || null}
        onClick={this.handleClick}
      >
        {typeof children === 'string' ? (
          <Text color={color}>{children}</Text>
        ) : (
          children
        )}
      </button>
    )
  }
}

Button.defaultProps = {
  type: 'button',
  disabled: false,
  color: 'white',
  backgroundColor: 'purple',
  disabledColor: 'darkgray',
  disabledBackgroundColor: 'lightGray',
}

const mapThemeToProps = (theme, props) => ({
  backgroundColor: props.backgroundColor || theme.primary,
})

export default withTheme(mapThemeToProps)(Button)
