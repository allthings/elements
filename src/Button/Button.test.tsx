import * as React from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'
import ThemeProvider from '../ThemeProvider'

describe('Button component', () => {
  test('Renders without anything', () => {
    const tree = renderer.create(<Button>Hi</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Renders a callback anything', () => {
    const tree = renderer.create(<Button onClick={_ => _}>Hi</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('doesnt render the callback if disabled', () => {
    const tree = renderer
      .create(
        <Button disabled onClick={_ => _}>
          Hi
        </Button>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Renders in different colors', () => {
    const tree = renderer
      .create(
        <Button
          name="asd"
          color="primary"
          backgroundColor="warn"
          onClick={_ => _}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Change color to green', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={{ warn: 'green' }}>
          <Button name="asd" backgroundColor="warn" onClick={_ => _} />
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
