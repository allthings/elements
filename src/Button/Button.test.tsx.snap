// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Change color to green 1`] = `
.css-0,
[data-css-0] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.css-1,
[data-css-1] {
  position: relative;
  padding: 10px 16px;
  border-radius: 2px;
  border: none;
  user-select: none;
  outline: none;
  color: white;
  background: green;
  cursor: pointer;
  transition: 250ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: 250ms ease-in-out;
  -moz-transition: 250ms ease-in-out;
}

.css-1:focus,
[data-css-1]:focus {
  outline: none;
}

.css-1:hover,
[data-css-1]:hover {
  color: white;
  background: green;
}

<button
  data-css-0=""
  data-css-1=""
  name="asd"
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button component Renders a callback anything 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  padding: 10px 16px;
  border-radius: 2px;
  border: none;
  user-select: none;
  outline: none;
  color: white;
  background: #3598db;
  cursor: pointer;
  transition: 250ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: 250ms ease-in-out;
  -moz-transition: 250ms ease-in-out;
}

.css-0:focus,
[data-css-0]:focus {
  outline: none;
}

.css-0:hover,
[data-css-0]:hover {
  color: white;
  background: #227dbf;
}

.css-3,
[data-css-3] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<button
  data-css-3=""
  data-css-0=""
  name="button"
  onClick={[Function]}
  type="button"
>
  Hi
</button>
`;

exports[`Button component Renders in different colors 1`] = `
.css-0,
[data-css-0] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.css-1,
[data-css-1] {
  position: relative;
  padding: 10px 16px;
  border-radius: 2px;
  border: none;
  user-select: none;
  outline: none;
  color: #3598db;
  background: #e84c3d;
  cursor: pointer;
  transition: 250ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: 250ms ease-in-out;
  -moz-transition: 250ms ease-in-out;
}

.css-1:focus,
[data-css-1]:focus {
  outline: none;
}

.css-1:hover,
[data-css-1]:hover {
  color: #3598db;
  background: #db2d1a;
}

<button
  data-css-0=""
  data-css-1=""
  name="asd"
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button component Renders without anything 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  padding: 10px 16px;
  border-radius: 2px;
  border: none;
  user-select: none;
  outline: none;
  color: white;
  background: #3598db;
  cursor: pointer;
  transition: 250ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: 250ms ease-in-out;
  -moz-transition: 250ms ease-in-out;
}

.css-0:focus,
[data-css-0]:focus {
  outline: none;
}

.css-0:hover,
[data-css-0]:hover {
  color: white;
  background: #227dbf;
}

.css-3,
[data-css-3] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<button
  data-css-3=""
  data-css-0=""
  name="button"
  type="button"
>
  Hi
</button>
`;

exports[`Button component doesnt render the callback if disabled 1`] = `
.css-0,
[data-css-0] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.css-1,
[data-css-1] {
  position: relative;
  padding: 10px 16px;
  border-radius: 2px;
  border: none;
  user-select: none;
  outline: none;
  color: white;
  background: lightGrey;
  cursor: not-allowed;
  transition: 250ms ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-transition: 250ms ease-in-out;
  -moz-transition: 250ms ease-in-out;
}

.css-1:focus,
[data-css-1]:focus {
  outline: none;
}

.css-1:hover,
[data-css-1]:hover {
  color: white;
  background: lightGrey;
}

<button
  data-css-0=""
  data-css-1=""
  name="button"
  type="button"
>
  Hi
</button>
`;
